includes/class-wcs-att-cart.php

```diff
@@ -456,7 +504,63 @@ public static function is_supported_product_type( $arg ) {
 		return $is_supported;
 	}
 
+	/**
+	 * Validates the subscription schemes applied on cart items.
+	 */
+	public static function check_applied_subscription_schemes() {
+
+		foreach ( WC()->cart->get_cart() as $cart_item_key => $cart_item ) {
+
+			$scheme_to_apply = self::get_subscription_scheme( $cart_item );
+			$applied_scheme  = WCS_ATT_Product_Schemes::get_subscription_scheme( $cart_item[ 'data' ] );
+
+			if ( $scheme_to_apply !== $applied_scheme && ! isset( $cart_item[ 'subscription_renewal' ] ) ) {
+
+				$available_schemes  = WCS_ATT_Product_Schemes::get_subscription_schemes( $cart_item[ 'data' ] );
+				$has_forced_schemes = WCS_ATT_Product_Schemes::has_forced_subscription_scheme( $cart_item[ 'data' ] );
+				$has_scheme_options = sizeof( $available_schemes ) > 1 || ( 1 === sizeof( $available_schemes ) && false === $has_forced_schemes );
+
+				// The product was purchased as a subscription...
+				if ( $scheme_to_apply ) {
+
+					// ...and the purchased scheme does not exist anymore...
+					if ( ! in_array( $scheme_to_apply, $available_schemes ) ) {
+
+						// If options exist, request a change.
+						if ( $has_scheme_options ) {
+							$prompt = __( 'Please choose a new subscription plan.', 'woocommerce-subscribe-all-the-things' );
+						// If not, show a blocking notice. Later on, we can add a button here to provide an opportunity to the user to resolve the problem.
+						} else {
+							$prompt = '';
+						}
+
+						wc_add_notice( sprintf( __( 'The &quot;%1$s&quot; subscription that you signed up for is no longer available. %2$s', 'woocommerce-subscribe-all-the-things' ), $cart_item[ 'data' ]->get_name(), $prompt ), 'error' );
 
+					// ...or a dev misbehaved and deserves some bad karma.
+					} else {
+						error_log( sprintf( 'Incorrect subscription scheme applied to cart item %s (%s). Scheme to apply: "%s". Applied scheme: "%s".' ), $cart_item_key, $cart_item[ 'data' ]->get_name(), var_export( $scheme_to_apply, true ), var_export( $applied_scheme, true ) );
```

Comment:
@thenbrent Added some basic logging at https://github.com/Prospress/woocommerce-subscribe-all-the-things/pull/173/files#diff-e5bc18f1662c7e2ec4e4533bf5f1dac3R149

Note we don't log the Order / Item ID there because the CRUD object hasn't been saved yet. I thought the timestamp would be enough for any future investigations, let me know if you think otherwise.